util/helm/cmd.go

```diff
@@ -342,19 +342,32 @@ type TemplateOpts struct {
 	SkipCrds    bool
 }
 
-var (
-	re                 = regexp.MustCompile(`([^\\]),`)
-	apiVersionsRemover = regexp.MustCompile(`(--api-versions [^ ]+ )+`)
-)
-
 func cleanSetParameters(val string) string {
 	// `{}` equal helm list parameters format, so don't escape `,`.
 	if strings.HasPrefix(val, `{`) && strings.HasSuffix(val, `}`) {
 		return val
 	}
-	return re.ReplaceAllString(val, `$1\,`)
+
+	var result strings.Builder
+	var prevR rune
+	for _, r := range val {
+		if r == ',' {
+			if prevR != '\\' {
```

Comment:
```suggestion
			if prevR != `\` {
```

More readable?

Reply:
Thanks for the suggestion. However, for that code, there is a bug where the single quotes are not closed because `\` escapes ` in GO.
![image](https://github.com/user-attachments/assets/f0ec45a1-a579-4b48-a26d-4b0b893889f8)

Reply:
Looks like you're using a single quote, I was looking for a backtick

<img width="157" alt="image" src="https://github.com/user-attachments/assets/9c9b58ff-bb67-490f-b3c7-0d2a3ab46b55">

Reply:
Ah, I get it.
But when I use backticks, the string comparison fails like in the picture. 
It seems to fail the comparison with rune type, which is the type of prevR, because it is a string when expressed using backticks. 
![image](https://github.com/user-attachments/assets/4521b28c-98e5-46df-b9f1-11c86b5c1742)

Reply:
Ah makes sense. Nevermind then!